etcdserver/api/membership/cluster.go

```diff
@@ -262,7 +280,7 @@ func (c *RaftCluster) ValidateConfigurationChange(cc raftpb.ConfChange) error {
 	}
 	switch cc.Type {
 	case raftpb.ConfChangeAddNode, raftpb.ConfChangeAddLearnerNode:
-		if members[id] != nil {
+		if members[id] != nil && !members[id].IsLearner {
```

Comment:
Does this mean we can add multiple learners with same ID?

Reply:
I noticed it is ok for the raft, raft will drop the request. otherwise it needs to refactor the `ValidateConfigurationChange`. put more code check code to it.

Reply:
You are right, raft will drop the request. Not sure if the server should rely on that though. Maybe for now we can add a comment here explaining, otherwise from reading this part of the code, readers might get confused.